parser/feed/src/main/java/de/danoeh/antennapod/parser/feed/PodcastIndexTranscriptParser.java

```diff
@@ -0,0 +1,191 @@
+package de.danoeh.antennapod.parser.feed;
+
+import android.util.Log;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.internal.StringUtil;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import de.danoeh.antennapod.model.feed.Transcript;
+import de.danoeh.antennapod.model.feed.TranscriptSegment;
+
+public class PodcastIndexTranscriptParser {
+
+    public static class PodcastIndexTranscriptSrtParser {
+
+        private static String TAG = "PodcastIndexTranscriptSrtParser";
+
+        public static Transcript parse(String str) {
+            Transcript transcript = new Transcript();
+
+            TranscriptSegment transcriptSegment;
+            List<String> lines = Arrays.asList(str.split("\n"));
+            Iterator<String> iter = lines.iterator();
+            String speaker = "";
+            StringBuffer body = new StringBuffer("");
+            String line;
+            String segmentBody = "";
+            long startTimecode = 0L;
+            long spanStartTimecode = 0L;
+            long endTimecode = 0L;
+            long span = 1000L;
+            long duration = 0L;
+
+            do {
+                line = iter.next();
```

Comment:
Doesn't this crash if there are no lines? Why does this need to be `do...while`? instead of a simple `while`?

Reply:
switched to `while`